### PR TITLE
Snowflake and BigQuery destination V2

### LINKED ISSUE #29681: Dev images of v2.0.0 connectors to test

### PR DESCRIPTION
**Pull request by evantahler**
### 💀 DO NOT MERGE ⚠️ 

Closes https://github.com/airbytehq/airbyte/issues/29681

Publish multiple connectors at once with
```
--name=destination-bigquery --name=destination-snowflake 
--pre-release
CHANGE THE BRANCH
```

To make a local registry:
1. download https://storage.googleapis.com/prod-airbyte-cloud-connector-metadata-service/registries/v0/oss_registry.json
2. replace the values you want
3. use it locally via `CONNECTOR_REGISTRY_BASE_UR=path/to/json.json`

Prereleases:

| Date | BigQuery | Snowflake | Run | 
|----| ------|-----------|-----|
| 2023-08-21 | [airbyte/destination-bigquery:2.0.0-dev.b82ab8b780](https://hub.docker.com/r/airbyte/destination-bigquery/tags) | [airbyte/destination-snowflake:3.0.0-dev.b82ab8b780](https://hub.docker.com/r/airbyte/destination-snowflake/tags) | https://github.com/airbytehq/airbyte/actions/runs/5931864372 | 
| 2023-08-21 | [airbyte/destination-bigquery:2.0.0-dev.195ce56d4e](https://hub.docker.com/r/airbyte/destination-bigquery/tags) | [airbyte/destination-snowflake:3.0.0-dev.195ce56d4e](https://hub.docker.com/r/airbyte/destination-snowflake/tags) | https://github.com/airbytehq/airbyte/actions/runs/5933074788|  
| 2023-08-23 | [airbyte/destination-snowflake:3.0.0-dev.912e60c225](https://hub.docker.com/r/airbyte/destination-snowflake/tags) | [airbyte/destination-bigquery:2.0.0-dev.912e60c225](https://hub.docker.com/r/airbyte/destination-bigquery/tags) | https://github.com/airbytehq/airbyte/actions/runs/5955875481 |

### CODE REVIEW

File: docs/integrations/destinations/snowflake.md [hunk 1]
```diff
@@ -270,7 +270,8 @@ Otherwise, make sure to grant the role the required permissions in the desired n
...
-|:----------------|:-----------|:-----------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------|
+| :-------------- | :--------- | :--------------------------------------------------------- | :-------------------------------------------------------------------------------------------------------------------------------------------------- |
+| 3.0.0           | 2023-08-27 | [xxx](https://github.com/airbytehq/airbyte/pull/xxx)       | Destinations V2                                                                                                                                     |
 | 2.0.0           | 2023-08-09 | [\#28894](https://github.com/airbytehq/airbyte/pull/29236) | Remove support for Snowflake GCS/S3 loading method in favor of Snowflake Internal staging                                                           |
```
Comment (evantahler) [inline docs/integrations/destinations/snowflake.md:275]: Of note: Snowflake + Destinations V2 is actually "v3.0.0"

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/TypingAndDedupingFlag.java [hunk 2]
```diff
@@ -10,7 +10,9 @@
 public class TypingAndDedupingFlag {
 
   public static boolean isDestinationV2() {
-    return DestinationConfig.getInstance().getBooleanValue("use_1s1t_format");
+    //TODO: Refactor this whole class away
+    // return DestinationConfig.getInstance().getBooleanValue("use_1s1t_format");
+    return true;
   }
```
Comment (evantahler) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/TypingAndDedupingFlag.java:16]: This is smallest possible change to 'force' V2 destinations to be enabled.

File: airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json [hunk 3]
```diff
@@ -207,17 +207,11 @@
...
       },
-      "use_1s1t_format": {
-        "type": "boolean",
-        "description": "(Early Access) Use <a href=\"https://docs.airbyte.com/understanding-airbyte/typing-deduping\" target=\"_blank\">Destinations V2</a>.",
-        "title": "Use Destinations V2 (Early Access)",
-        "order": 7
-      },
       "raw_data_dataset": {
         "type": "string",
         "description": "(Early Access) The dataset to write raw tables into",
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json:212]: remove the `early access` from these options

File: airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json [hunk 4]
```diff
@@ -161,17 +161,11 @@
...
       },
-      "use_1s1t_format": {
-        "type": "boolean",
-        "description": "(Beta) Use <a href=\"https://github.com/airbytehq/airbyte/issues/26028\" target=\"_blank\">Destinations V2</a>. Contact Airbyte Support to participate in the beta program.",
-        "title": "Use Destinations V2 (Early Access)",
-        "order": 10
-      },
       "raw_data_schema": {
         "type": "string",
         "description": "(Beta) The schema to write raw tables into",
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json:166]: and also from these

**Review by edgao — COMMENTED** [review]
do you also need to update [run_with_normalization.sh](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-java/run_with_normalization.sh#L36) to never run normalization? Or is that covered by the metadata change

also I think a lot of DATs are going to fail here, since they make assumptions about normalized tables existing.

**Comment by evantahler** [comment]
> do you also need to update [run_with_normalization.sh](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-java/run_with_normalization.sh#L36) to never run normalization? Or is that covered by the metadata change

Good point! The platform should be good about not running a normalization container now that the destinations no longer have a normalizationConfig metadata (https://github.com/airbytehq/airbyte-platform-internal/pull/8314), but the normal READ container might still get the `use_1s1t_format` option from an old connection's config file.

**Comment by edgao** [comment]
I just realized that we should make a second copy of that script, if we want to do beta periods for other connectors (so we would have an always-dv2 script, and a script that supports both modes). Problem for next week though!

**Comment by evantahler** [comment]
You know what?  The fine folks on Connector Ops already have us covered! 

https://github.com/airbytehq/airbyte/blob/evan/bigquery-snowflake-v2/airbyte-ci/connectors/pipelines/pipelines/actions/environments.py#L840-L849

The build script only uses the /airbyte/run_with_normalization.sh entry point if `context.connector.supports_normalization and DESTINATION_NORMALIZATION_BUILD_CONFIGURATION[context.connector.technical_name]["supports_in_connector_normalization"]`.  These new updates to the metadata don't pass those checks any more

**Comment by evantahler** [comment]
Closed in favor of https://github.com/airbytehq/airbyte/pull/29683

**Pull request closed without merging**